addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/handler/AirQualityHandler.java

```diff
@@ -0,0 +1,303 @@
+            if (value == null) {
+                state = UnDefType.UNDEF;
+            } else if (value instanceof Calendar) {
+                state = new DateTimeType((Calendar) value);
+            } else if (value instanceof Integer) {
+                state = new DecimalType(BigDecimal.valueOf(((Integer) value).longValue()));
+            } else if (value instanceof String) {
+                state = new StringType(value.toString());
+            } else {
+                logger.warn("Update channel {}: Unsupported value type {}", channelId,
+                        value.getClass().getSimpleName());
+            }
+            logger.debug("Update channel {} with state {} ({})", channelId, (state == null) ? "null" : state.toString(),
+                    (value == null) ? "null" : value.getClass().getSimpleName());
+
+            // Update the channel
+            if (state != null) {
+                updateState(channelId, state);
+            }
+        }
+    }
+
+    /**
+     * Get new data from aqicn.org service
+     *
+     * @return {boolean}
+     */
+    private synchronized boolean updateAirQualityData() {
+        AirQualityConfiguration config = getConfigAs(AirQualityConfiguration.class);
+        aqiData = getAirQualityData(StringUtils.trimToEmpty(config.location));
+        return aqiData != null;
+    }
+
+    /**
+     * Request new air quality data to the aqicn.org service
+     *
+     * @param location geo-coordinates from config
+     *
+     * @return the air quality data object mapping the JSON response or null in case of error
+     */
+    private AirQualityJsonResponse getAirQualityData(String location) {
+        AirQualityJsonResponse result = null;
+        boolean resultOk = false;
+        String errorMsg = null;
+
+        try {
+
+            // Build a valid URL for the aqicn.org service
+            AirQualityConfiguration config = getConfigAs(AirQualityConfiguration.class);
+
+            String geoStr = location.replace(" ", "").replace(",", ";").trim();
+
+            String urlStr = URL.replace("%APIKEY%", StringUtils.trimToEmpty(config.apikey));
+            urlStr = urlStr.replace("%LOCATION%", geoStr);
+            logger.debug("URL = {}", urlStr);
+
+            // Run the HTTP request and get the JSON response from aqicn.org
+            URL url = new URL(urlStr);
+            URLConnection connection = url.openConnection();
+            String repsonse = IOUtils.toString(connection.getInputStream());
+            logger.debug("aqiData = {}", repsonse);
+
+            // Map the JSON response to an object
+            result = gson.fromJson(repsonse, AirQualityJsonResponse.class);
+            if (result == null) {
+                errorMsg = "no data returned";
+            } else if (result.getStatus() == "error") {
+                errorMsg = result.getData().toString();
+            } else {
+                resultOk = true;
+
+                if (result.getData() == null) {
```

Comment:
Maybe change this to "result.getData() != null" and only then set resultOk to true? the if clause is slightly confusing 😄